lib/modules/manager/terragrunt/extract.ts

```diff
@@ -10,13 +10,17 @@ import {
 
 const dependencyBlockExtractionRegex = regEx(/^\s*(?<type>[a-z_]+)\s+{\s*$/);
 const contentCheckList = ['terraform {'];
+const includeBlockCheck = 'include "';
 
 export function extractPackageFile(
   content: string,
   packageFile?: string,
 ): PackageFileContent | null {
   logger.trace({ content }, `terragrunt.extractPackageFile(${packageFile!})`);
   if (!checkFileContainsDependency(content, contentCheckList)) {
+    if (content.includes(includeBlockCheck)) {
+      return { deps: [] };
+    }
```

Comment:
If the `include "` string will always be present at the beginning of the file. A regex check like `/^include\s"/` or `content.startsWith('include "')` is more suitable here.